src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -41,6 +47,29 @@ void VulnerabilityScannerFacade::start(
         auto& policyManager = PolicyManager::instance();
         policyManager.initialize(configuration);
 
+        m_stateDB = std::make_unique<Utils::RocksDBWrapper>(VD_STATE_QUEUE_PATH);
+        std::string lastState;
+        m_stateDB->get("previous_config", lastState);
+
+        if (!lastState.empty())
+        {
+            if (lastState == "no" && policyManager.isVulnerabilityDetectionEnabled())
+            {
+                logInfo(WM_VULNSCAN_LOGTAG, "Vulnerability scanner restarted");
+                m_stateDB->put("previous_config", "yes");
+                m_shouldClean.store(true);
+            }
+            else if (lastState == "yes" && !policyManager.isVulnerabilityDetectionEnabled())
+            {
+                m_stateDB->put("previous_config", "no");
+                m_shouldClean.store(true);
```

Comment:
This might not have effect because the module is disabled.

Reply:
Remove it!